### DISCUSSION TITLE
Homelab portal?

### DISCUSSION THREAD

**Discussion by joshmouch** [microsoft/aspire#4152] [General]
I was just wondering if Aspire.NET might be a good fit for a "Home Lab/Smart Home Server" type dashboard.
The orchestration is obviously big, but so is the already-built dashboard that natively collects and displays URLs, statuses, and logs for all the different services that are in use in a Smart Home/Homelab (e.g. Homeassistant, Node-red, NetworkControllerSoftware, Metrics, MQTT broker, Frigade, Camera recording and AI services, Matter and BLE and Zigbee and Z-Wave and SmartThings integration proxies, etc.).

Since this isn't a production environment, there's benefit to updating the "deployed orchestration" immediately. So, what I'm wondering is if there's any gotchas to just running something like `dotnet watch MyHomeLab.AppHost.sln` on server boot (maybe from a system-startup-initiated Windows Service with `Local System` credentials) and letting Aspire.NET rebuild itself on any detected changes in code (i.e. new docker container or what have you). 

`dotnet watch MyHomeLab.AppHost.sln`

The reason I'm asking is because that `Aspire.Hosting.Dcp` service is a bit of a black box.  I believe its GitHub is private to Microsoft.

So, I guess my question is really: are there any reasons that the binaries in the `Aspire.Hosting.Dcp` namespace can't run without a windows login or as Local System or as a Windows Service or for months on end or be restarted randomly with a `dotnet watch` file-change-trigger or abruptly stopped without corruption (e.g. persistent mapped volume)?

**Comment by 3GDXC** [comment]
@joshmouch not sure what/where your looking or if I've misunderstood; have you seen src/Aspire.Hosting/Dcp
